Sources/MapboxNavigation/NavigationOptions.swift

```diff
@@ -47,6 +47,48 @@ open class NavigationOptions: NavigationCustomizable {
      */
     open var predictiveCacheOptions: PredictiveCacheOptions?
     
+    
+    /**
+     To add docs for cases
+     */
+    public enum TileStoreLocation {
```

Comment:
I am concerned about this thing. From one perspective, since Map and Navigator have slightly different use-cases (Map may have no TileStore) using `URL?` as a setting does not cover all possible cases. In this case `nil` for Nav == default, while `nil` for map == no tile store.
Introducing this enum construct allows fine control and convenience usage like:
```
options.tileStoreLocation = .custom(url) // to pass user-provided location, or
options.tileStoreLocation = .isolated(.default, noStorage) // to disable Map Tile storage while keeping Nav at default
```
On the other hand - this setting is not reusable while manipulating bare `NavigationView` or `NavigationService`. It is also very sensitive to initialisation order. If User wants to create his custom `NavigationService` to pass it in `NavigationOptions` - he has to also take care of configuring a tile storage parameter.

Reply:
And yes, naming is an open question

Reply:
An enumeration with associated values seems reasonable to me.

> this setting is not reusable while manipulating bare NavigationView or NavigationService

Not sure I follow – we could modify those classes to accept a TileStoreLocation too, right?

Reply:
> Not sure I follow – we could modify those classes to accept a TileStoreLocation too, right?

In this case we should frame it outside `NavigationOptions` to avoid ugly`NavigationService.init` definition, and it makes `TileStoreLocation` a standalone type which I'd like to avoid. Also, in terms of bare `NavigationMapView` plain `TileStore?` is a more robust way for picking storage location.